addons/loyalty/models/loyalty_program.py

```diff
@@ -577,3 +577,12 @@ def _get_template_values(self):
                 })]
             },
         }
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            if (vals.get('trigger_product_ids') is not None and
+                    vals['program_type'] not in ['gift_card', 'ewallet']):
+                del vals['trigger_product_ids']
```

Comment:
This could use a short but clear comment on why we're doing that. The blame history should only be there as a last resort for understanding why a piece of code exists.